pubtools/_quay/operator_pusher.py

```diff
@@ -395,6 +398,33 @@ def get_existing_index_images(self, quay_client):
                 )
         return list(set(current_index_images))
 
+    @log_step("Verify bundles presence")
+    def ensure_bundles_present(self):
+        """
+        Make sure bundles are present in Quay.
+
+        Wait until pushed images become present in Quay,
+        and return False if too much time has passed.
+        """
+        bundles = [self.public_bundle_ref(i) for i in self.push_items]
+        for bundle in bundles:
+            registry = bundle.split("/", 1)[0]
+            username, password = get_basic_auth(registry)
```

Comment:
How are these values added to the config file? Aren't they somewhere in the target settings? Is it possible that they wouldn't be in the config file?

Reply:
Bundle images are from registry(.qa|.stage).redhat.io. Auths of them are added to config.json by pub-playbook. 

Quay-mig targets have source_quay_host set to one of registry(.qa|.stage).redhat.io, and source_quay_user, source_quay_password are set with its auth. However, source_quay_host is quay.io for regular quay targets, and the username, password are quay auth.